src/org/zaproxy/zap/extension/pscanrulesAlpha/CacheableScanner.java

```diff
@@ -407,7 +407,14 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 								lifetimeFound = true;
 								lifetimesFound++;
 								//get the portion of the string after "maxage="
-								lifetime = Long.parseLong(directiveToken.substring("max-age=".length()));
+								// Split on comma and use 0th item in case there weren't spaces: 
+								// Cache-Control: max-age=7776000,private
+								try {
+									lifetime = Long.parseLong(directiveToken.split(",")[0].substring("max-age=".length()));
+								} catch (NumberFormatException nfe) {
+									lifetimeFound = false;
```

Comment:
We should also decrement `lifetimesFound`? We could move those (also `freshEvidence = ...`) to after the (successful) parse instead?

Reply:
OK.